internal/x11/win/frame.go

```diff
@@ -579,17 +585,40 @@ func (f *frame) mouseRelease(x, y int16) {
 		return
 	}
 
-	obj := wm.FindObjectAtPixelPositionMatching(int(relX), int(relY), f.canvas,
-		func(obj fyne.CanvasObject) bool {
-			_, ok := obj.(fyne.Tappable)
-			return ok
-		},
-	)
-
-	if obj == nil {
+	if f.ctx != nil && f.cancelFunc != nil {
+		f.doubleClick = true
+		f.cancelFunc()
 		return
 	}
-	obj.(fyne.Tappable).Tapped(&fyne.PointEvent{})
+	go func() {
+		f.ctx, f.cancelFunc = context.WithDeadline(context.TODO(), time.Now().Add(time.Millisecond*300))
+		defer f.cancelFunc()
+		select {
+		case <-f.ctx.Done():
+			if f.doubleClick == true {
+				if f.client.Maximized() {
+					f.client.Unmaximize()
+				} else {
+					f.client.Maximize()
+				}
+			} else {
+				obj := wm.FindObjectAtPixelPositionMatching(int(relX), int(relY), f.canvas,
```

Comment:
Keeping this inline makes the code much more complex to read.
It looks like some functions should be factored out here, as there are many scopes:
*) top level - handles a mouse release
*) go routine - runs a timer to detect type of mouse release
*) first fork - has what to do if double tapped
*) second fork - the previous tap handling code

That's a lot of different scopes for one method.